src/instantsend/instantsend.h

```diff
@@ -85,11 +84,18 @@ class CInstantSendManager final : public instantsend::InstantSendSignerParent
 
 public:
     explicit CInstantSendManager(CChainLocksHandler& _clhandler, CChainState& chainstate, CQuorumManager& _qman,
-                                 CSigningManager& _sigman, CSigSharesManager& _shareman, CSporkManager& sporkman,
-                                 CTxMemPool& _mempool, const CMasternodeSync& mn_sync, bool is_masternode,
-                                 bool unitTests, bool fWipe);
+                                 CSigningManager& _sigman, CSporkManager& sporkman, CTxMemPool& _mempool,
+                                 const CMasternodeSync& mn_sync, bool unitTests, bool fWipe);
     ~CInstantSendManager();
 
+    void ConnectSigner(gsl::not_null<instantsend::InstantSendSigner*> signer)
+    {
+        // Prohibit double initialization
+        assert(m_signer == nullptr);
+        m_signer = signer;
+    }
+    void DisconnectSigner() { m_signer = nullptr; }
```

Comment:
I believe here's possible race for `m_signer`.

what will happen if `DisconnectSigner` will be called during `CInstantSendManager::ProcessInstantSendLock`? especially assuming `DisconnectSigner` is a public method